src/main/resources/templates/admin/post/show.ftl

```diff
@@ -3,15 +3,25 @@
 <div class="container">
 
     <div class="row mt-4 d-flex align-items-center  justify-content-end">
-        <@security.authorize access="hasRole('ROLE_WRITER')">
+        <@security.authorize access="!hasRole('ROLE_ADMIN')">
```

Comment:
I believe, it's better to avoid negation in such cases, because it can be more roles added in the future and such places will be the first candidates to introduce bugs.
Just use `hasAnyRole('ROLE_WRITER', 'ROLE_EDITOR')` in all places like this.